### ISSUE TITLE
`imProc.astrometry.astrometryQualityData` error

### ISSUE THREAD

**Issue by EastEriq** [EranOfek/AstroPack#170]
```
 R=imProc.astrometry.astrometryQualityData(AI)
Unable to perform assignment because the size of the left side is 1-by-1 and the size
of the right side is 0-by-1.

Error in fminsearch (line 216)
fv(:,1) = funfcn(x,varargin{:});

Error in celestial.coo.boundingCircle (line 48)
[BestCoo, BestRadius] = fminsearch(@radiusForCenter,[MidLon, MidLat], Options);

Error in AstroCatalog/boundingCircle (line 696)
                        [BestCoo, BestRadius] = celestial.coo.boundingCircle(X, Y);   % [radians]

Error in imProc.match.returnMatched_catsHTM (line 100)
            [CircX, CircY, CircR] = CatIn(Iobj).boundingCircle('OutUnits','rad', 'CooType','sphere');

Error in imProc.astrometry.astrometryQualityData (line 153)
        [MatchedRefCat(Iobj), ResInd(Iobj), CatH] = imProc.match.returnMatched_catsHTM(Obj(Iobj), Args.CatName,...

```
AI is a processed PTF image ( /raid/enrico/PTF/2ndTake/images/level1/proc/2012/01/07/f12/c0/p5/v1/PTF_201201075042_i_p_scie_t120602_u011065103_f12_p015199_c00.fits), with the code
```
T=readIVOtable('/raid/enrico/PTF/2ndTake/M81.tbl');
ff=fullfile('/raid/enrico/PTF/2ndTake/images/level1',T.pfilename);
AI=AstroImage(ff(5));
AI.populateWCS; %not automatical, https://github.com/EranOfek/AstroPack/issues/151
imProc.calib.gainCorrect(AI);

% ad hoc corrections for defects of ccd 0
if AI.HeaderData.getVal('CCDID')==0
    AI.Image=adHocTreatmentPTFccd0(AI.Image);
end

imProc.calib.equaliseLineMean(AI);

AI.MaskData.Data=mapPTF2oursMask(MaskImage(strrep(ff{1},'scie','mask')));

imProc.background.background(AI, 'SubSizeXY',[128 128]);
imProc.mask.maskHoles(AI);
imProc.mask.maskSourceNoise(AI);

%  no imProc equivalent for masking bad columns, do it in steps
%  threshold is a bit delicate. I would prefer a more robust method,
%    marking perhaps only the bad part of the column
% mask=imUtil.background.flag_badcol(AI.Image,'Dim',1,'CollapseFun','std','Threshold',50);

imProc.sources.findMeasureSources(AI,'RemoveBadSources',true,'RemoveEdgeDist',3);
imProc.psf.constructPSF(AI); % a few hundred Nsrc
[~,Sources]=imProc.sources.psfFitPhot(AI) % thousands Sources
astrometry=imProc.astrometry.astrometryRefine(AI)

% file #3, bad column 461, lots of false sources

ds9(AI,'zoom','to fit')
%ds9.plotXY([Sources.X(Sources.ConvergeFlag),Sources.Y(Sources.ConvergeFlag)])
ds9.plotXY([Sources.X,Sources.Y])
%ds9.plotXY([Sources.X(Sources.Chi2<100),Sources.Y(Sources.Chi2<100)])
```

**Comment by EranOfek** [comment]
Note that AI.poopualteWCS is not needed anymore - happens automatically.

**Comment by EranOfek** [comment]
The problem is that the RA/Dec were not propogatyed to the catalog.
To fix this:
1. It will now will fain with error: "Can not find RA/Dec coordinates in catalog"
2. in astrometryRefine I added a new option: 'AddCoo2Cat' which will do it (note that right now the default is false)
3. Instead you can directly use:
```
AI=imProc.astrometry.addCoordinates2catalog(AI)
```

**Comment by EastEriq** [comment]
I'm using `[W,AI]=imProc.astrometry.astrometryRefine(AI,'AddCoo2Cat',true)`, but now I get
```
Error using AstroCatalog/boundingCircle
Can not find RA/Dec coordinates in catalog

Error in imProc.calib.photometricZP (line 229)
                    [RA, Dec, CircleRadius] = boundingCircle(Cat, 'OutUnits','rad', 'CooType','sphere');
```
when I subsequently call `[AI, ZP, PhotCat] =imProc.calib.photometricZP(AI)`

**Comment by EastEriq** [comment]
And if I prepend the `AI=imProc.astrometry.addCoordinates2catalog(AI)` to that call to `photometricZP()`, I get
```
Index exceeds the number of array elements. Index must not exceed 0.

Error in VO.search.search_sortedlat_multiNearest (line 67)
Ilowhigh = double(Inear(Ilat));

Error in imProc.match.matchReturnIndices (line 152)
        [IndTable, CatFlagNearest, CatFlagAll, IndInObj2] = VO.search.search_sortedlat_multiNearest(Coo1,...

Error in imProc.calib.photometricZP (line 260)
            ResMatch = imProc.match.matchReturnIndices(PhotCat(Iobj), Cat, 'Radius',Args.Radius,...
```
Great.

**Comment by EastEriq** [comment]
Guess what is `DistFun` at line 152 of ` imProc.match.matchReturnIndices`. Again #156 . ^%#&^@#!!!

**Comment by EastEriq** [comment]
but besides line 237 of `imProc.match.matchReturnIndices` retrieves an empty `PhotCat(1)`, which is the cause of this error. Give me a break.

**Comment by EastEriq** [comment]
The empty catalog comes from this call: https://github.com/EranOfek/AstroPack/blob/9e08b02e19594ffedc43d1575671553aeb5378c0/matlab/image/%2BimProc/%2Bcat/getAstrometricCatalog.m#L102
which calls `cone_search`. Guess which function uses `cone_search`?
But that anyway seems not the problem. `boundingCircle` at line 229 gets a very small `CircleRadius`, I suspect. Maybe it is an unit issue. At that point, `Cat.ColUnits` are mostly empty, notably columns 52 and 53 for Ra and Dec.

**Comment by EranOfek** [comment]
I suspect you are right - this is a units problem.

**Comment by EastEriq** [comment]
Opened issue #213 separately for this

**Comment by EastEriq** [comment]
The latter problems solved by commit 9ce248d62fc.
